rfcs/2021-07-19-8216-multiple-pipelines.md

```diff
@@ -0,0 +1,260 @@
+  }
+
+  fn transform(&mut self, output: &mut Vec<Event>, event: Event);
+}
+```
+
+This solution would required the core of vector and how it handles transforms but would avoid generatic dynamic transforms.
+It would probably also reduce the performance considering that each transform would have to check if it's part of a pipeline.
+
+## Doc-level Proposal
+
+With this pipeline definition, the following `vector.toml`
+
+```toml
+[sources.docker]
+type = "docker_logs"
+...
+
+[transforms.nginx_filter]
+input = ["docker"]
+type = "remap"
+source = '''
+  # do something that filter nginx logs from docker logs
+'''
+
+[transforms.nginx]
+input = ["nginx_filter"]
+type = "remap"
+source = '''
+  # do some transformations with the nginx logs
+'''
+
+[transforms.apache_filter]
+input = ["docker"]
+type = "remap"
+source = '''
+  # do something that filter apache logs from docker logs
+'''
+
+[transforms.apache]
+input = ["apache_filter"]
+type = "remap"
+source = '''
+  # do some transformations with the apache logs
+'''
+
+[sinks.ouput]
+inputs = ["apache", "nginx"]
+type = "console"
+...
+```
+
+can be split into the following `pipelines/frontend.toml`
+
+```toml
+[transforms.nginx_filter]
+input = ["docker"]
+type = "remap"
+source = '''
+  # do something that filter nginx logs from docker logs
+'''
+
+[transforms.nginx]
+input = ["nginx_filter"]
+type = "remap"
+source = '''
+  # do some transformations with the nginx logs
+'''
+
+[transforms.apache_filter]
+input = ["docker"]
+type = "remap"
+source = '''
+  # do something that filter apache logs from docker logs
+'''
+
+[transforms.apache]
+input = ["apache_filter"]
+type = "remap"
+source = '''
+  # do some transformations with the apache logs
+'''
+```
+
+and `vector.toml`
+
+```toml
+[sources.docker]
+type = "docker_logs"
+...
+
+[sinks.ouput]
+inputs = ["frontend.apache", "frontend.nginx"]
+type = "console"
+...
+```
+
+## Rationale
+
+This improves the readability of the `vector.toml` and add the possibility to share pipelines or deploy them with some tools like Terraform or Ansible without modifying the main configuration.
+In the future, this will allow us to replicate datadog's pipelines locally.
+
+## Drawbacks
+
+- Not yet able to use a pipeline in another pipeline, but this may come in further work.
+- First option: creating some hidden transforms and modifying the existing ones could introduce some bugs.
+
+## Alternatives
+
+- Updating the `Topology` to create those transforms at the topology building time in order to avoid modifying the inputs when building the configuration.
+
+## Outstanding Questions
+
+- Should we have some visibility attributes on the components in different pipelines in order to make them available to other components?
+- Should we have a main (or default) transform in a pipeline that would allow to use the pipeline's ID as a transform name?
+
+## Plan Of Attack
+
+1. Update the CLI to take the pipeline directory parameter
```

Comment:
I'd argue for exposing this configuration parameter last, or at least gate it behind a default-off feature flag so we don't confuse users.

Reply:
Is the purpose of making this configurable to get around OS user-level permissions?

Reply:
Oh no, I just want to avoid exposing something that isn't available / ready yet.